dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatMessageBase.cs

```diff
@@ -17,6 +17,11 @@ public abstract class ChatMessageBase
     /// </summary>
     public string Content { get; set; }
 
+    /// <summary>
+    /// If role is function, the name of the function that produced the content
+    /// </summary>
+    public string? Name { get; set; } = null;
```

Comment:
Surfacing @stephentoub 's suggestion from the issue:

> If we're not comfortable adding a name property to a chat message in support of this, we could also just add a dictionary to the message so that additional context can be provided and interpreted by the chat completion service however it likes:
https://github.com/microsoft/semantic-kernel/compare/main...stephentoub:semantic-kernel:fixfunctioncalling

Reply:
@gitri-ms I would not change Abstractions regarding FunctionCalling support, instead I would think of having a Specialized ChatHistory class for OpenAI where you can add that function. 
On the examples where we know you are dealing with a OpenAIModel, you can check if the ChatHistory is an OpenAIChatHistory and play with the detailed function data.

`SKChatMessage` of a `OpenAIChatHistory` will have the name.

Reply:
I understand the concern here, but if we don't have a way to support additional fields like this in the base object, then devs will have to downcast the ChatHistory whenever they want to interact with OpenAI features, which makes for a really clunky experience. Talking with @alliscode, we thought that @stephentoub's approach would be most flexible and future-proof, so I updated this PR to add an optional dictionary to `ChatMessageBase`.